lib/auth/auth_with_roles.go

```diff
@@ -5957,31 +5957,46 @@ func (a *ServerWithRoles) UpdateHeadlessAuthenticationState(ctx context.Context,
 
 // CreateAssistantConversation creates a new conversation entry in the backend.
 func (a *ServerWithRoles) CreateAssistantConversation(ctx context.Context, req *proto.CreateAssistantConversationRequest) (*proto.CreateAssistantConversationResponse, error) {
-	// TODO(jakule): Check if user has access to given conversation ID
+	if err := a.action(apidefaults.Namespace, types.KindAssistant, types.VerbCreate); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
 	return a.authServer.CreateAssistantConversation(ctx, req)
 }
 
 // GetAssistantConversations returns all conversations started by a user.
 func (a *ServerWithRoles) GetAssistantConversations(ctx context.Context, request *proto.GetAssistantConversationsRequest) (*proto.GetAssistantConversationsResponse, error) {
-	// TODO(jakule): Check if user has access to given conversation ID
+	if err := a.action(apidefaults.Namespace, types.KindAssistant, types.VerbList); err != nil {
```

Comment:
In these get/update methods, should we allows users to read/update their own conversations? Like we use `currentUserAction` in some other RBAC checks to allow users update their own passwords or read their own access requests for example.

Reply:
This is a bit different case. Currently, users have only access to their own conversations.
We want to keep this feature enabled by default, but as an administrator, I think you should have the option to disable access for some users. In this case, we should disable access to all APIs, and users with disabled access should not able able to even list their own conversations.